dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,287 @@
+            .build()
+
+        // Then
+        assertThat(result.task(":$taskName")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @ParameterizedTest
+    @MethodSource("uploadTasks")
+    fun `M success W upload { checkProjectDependecies set to warn }`(taskName: String) {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_set_to_warn.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments(taskName)
+            .withPluginClasspath()
+            .build()
+
+        // Then
+        assertThat(result.task(":$taskName")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M not contain any uploadTasks W minifyNotEnabeld`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_minify_not_enabled.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("tasks")
+            .withPluginClasspath()
+            .build()
+
+        // Then
+        val uploadTask = result.output
+            .split("\n")
+            .firstOrNull { it.startsWith("upload") }
+        assertThat(uploadTask).isNull()
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun stubFile(destination: File, content: String) {
+        with(destination.outputStream()) {
+            this.write(content.toByteArray())
+        }
+    }
+
+    private fun stubGradleBuildFromResourceFile(resourceFilePath: String) {
+        javaClass.classLoader.getResource(resourceFilePath)?.file?.let {
+            File(it).copyTo(buildFile)
+        }
+    }
+
+    // endregion
+
+    companion object {
+        val ANDROID_MANIFEST_FILE_CONTENT = """
+            <?xml version="1.0" encoding="utf-8"?>
+            <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+                      package="com.example.variants">
+
+                <application
+                    android:allowBackup="true"
+                    android:supportsRtl="true">
+                    <activity android:name=".MainActivity">
+                        <intent-filter>
+                            <action android:name="android.intent.action.MAIN"/>
+
+                            <category android:name="android.intent.category.LAUNCHER"/>
+                        </intent-filter>
+                    </activity>
+                </application>
+
+            </manifest>
+        """.trimIndent()
+
+        const val SETTINGS_FILE_CONTENT = "rootProject.name = 'sample'"
+        val GRADLE_PROPERTIES_FILE_CONTENT = """
+           org.gradle.jvmargs=-Xmx2560m -XX:MaxPermSize=512m
+           android.useAndroidX=true
+           DD_API_KEY=fakekey
+        """.trimIndent()
+
+        @Suppress("unused")
+        @JvmStatic
+        fun uploadTasks(): Stream<Arguments> {
```

Comment:
in case of single method argument method source can just return a list of strings per https://junit.org/junit5/docs/5.3.0-RC1/api/org/junit/jupiter/params/provider/MethodSource.html, no need to use Stream + Argument

Reply:
Ah cool, didn't know that ;)